test/libsolidity/semanticTests/expressions/module_from_ternary_expression.sol

```diff
@@ -0,0 +1,14 @@
+==== Source: A ====
+contract C {
+}
+==== Source: B ====
+import "A" as M;
+
+contract C {
+    function f() public pure {
+        bool flag;
+        ((flag = true) ? M : M).C;
```

Comment:
Same as for tuples, you're not checking if the flag gets flipped. We need to know if side-effects are discarded or not in this kind of expression. Fortunately, unlike tuples, this case is safe to disallow if broken.

Reply:
Please also add a test case like this:

- `a.sol`
    ```solidity
    function f() pure returns (uint) {
        return 42;
    }
    ```
- `b.sol`
    ```solidity
    function f() pure returns (uint) {
        return 24;
    }
    ```
- `test.sol`
    ```solidity
    import "./a.sol" as A;
    import "./b.sol" as B;

    contract C {
        function f(bool a) public pure returns (uint) {
            return (a ? A : B).f();
        }
    }
    ```

Fortunately this triggers an error on 0.8.20:
```
Error: True expression's type module "a.sol" does not match false expression's type module "b.sol".
```

It if didn't it would actually make disallowing modules more questionable because this pattern does not seem completely unreasonable.

Reply:
Added.